### PR TITLE
feat(setup): Add Azure and AZS docs

### PR DESCRIPTION
**Pull request by ejizba**

### CODE REVIEW

**Comment by ejizba** [comment]
@rguthriemsft @dcaro @clguimanmsft PTAL

**Review by lwander — APPROVED** [review]
Nice!

**Comment by lwander** [comment]
I'll wait on more Azure-savy reviewers before merging

File: setup/providers/azure.md [hunk 1]
```diff
@@ -7,8 +7,77 @@ sidebar:
...
+
+Next, create a Service Principal and set environment variables based on the output:
+
+```bash
+az ad sp create-for-rbac --name "Spinnaker"
+APP_ID=<Insert App Id>
+TENANT_ID=<Insert Tenant Id>
+```
+
+> NOTE: You will need the App Key (also called password) when creating an account, but you will be prompted on standard input for that since it is sensitive data.
```
Comment (dcaro) [inline setup/providers/azure.md:38]: Add a note that the name "Spinnaker" used for the creation of the service principal must be unique in the Azure subscription.

File: setup/providers/azure.md [hunk 2]
```diff
@@ -7,8 +7,77 @@ sidebar:
...
+az account list-locations --query [].name
+RESOURCE_GROUP="Spinnaker"
+az group create --name $RESOURCE_GROUP --location <Insert Location>
+```
+
+Finally, create a Key Vault (where the vault name is globally unique) and add a default username/password:
+
+```bash
+VAULT_NAME=<Insert Vault Name>
+az keyvault create --enabled-for-template-deployment true --resource-group SpinnakerVault --name $VAULT_NAME
```
Comment (dcaro) [inline setup/providers/azure.md]: Change the resource group to "Spinnaker" to match the previous step.

**Review by dcaro — APPROVED** [review]
I have spotted two minor changes to make. The rest looks good to me.

File: setup/providers/azure.md (hunk 1 above)
Comment (ejizba) [reply setup/providers/azure.md:38]: Done

File: setup/providers/azure.md (hunk 2 above)
Comment (ejizba) [reply setup/providers/azure.md]: Nice catch! Meant to use the RESOURCE_GROUP environment variable

**Comment by lwander** [comment]
Ready to merge?

**Comment by ejizba** [comment]
Yep!

**Pull request merged**